### PR TITLE
NF-134 - upload files using Trader Service /file-transfer endpoint

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/FileTransferConnector.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  ): Future[FileTransferResult] =
+    http
+      .POST[TraderServicesFileTransferRequest, HttpResponse](url, fileTransferRequest)
+      .map(
+        response =>
+          FileTransferResult(
+            fileTransferRequest.upscanReference,
+            isSuccess(response),
+            response.status,
+            LocalDateTime.now(),
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/FileTransferConnector.scala:44]: OBSERVATION: This should come from `response. transferredAt`

Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/FileTransferConnector.scala:44]: Good point.  In fact, should we not just be passing the whole `FileTransferResult` returned from Trader Services?

Comment (crosslandwa) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/FileTransferConnector.scala:44]: Yes, that makes sense!

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/models/eis/TraderServicesFileTransferRequest.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  caseReferenceNumber: String,
+  applicationName: String,
+  upscanReference: String,
+  downloadUrl: String,
+  checksum: String,
+  fileName: String,
+  fileMimeType: String,
+  batchSize: Int,
+  batchCount: Int,
+  correlationId: Option[String] = None
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/models/eis/TraderServicesFileTransferRequest.scala:33]: OBSERVATION: I think we should remove provision of the `correlationId` until we are _sure_ we need it

As long as we are passing the `conversationID` (which needs to be provided to tie together a batch of file transfers) we can rely on the [downstream service generating](https://github.com/hmrc/trader-services-route-one/blob/master/app/uk/gov/hmrc/traderservices/controllers/FileTransferController.scala#L55) the `correlationID` (provided in the headers of the EIS request)

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/ClaimControllerSpec.scala [hunk 3]
```diff
@@ -78,9 +78,7 @@ class ClaimControllerSpec extends ControllerSpec with GuiceOneAppPerSuite with T
         when(mockClaimService.createClaim(any[EISCreateCaseRequest], anyString())(any())).thenReturn(
           Future.successful(eisSuccessResponse)
         )
-        when(mockFileTransferService.transfer(eisSuccessResponse.CaseID, Seq.empty)).thenReturn(
-          Future.successful(Seq.empty)
-        )
+        when(mockFileTransferService.transferFiles(any(), any(), any())(any())).thenReturn(Future.successful(Seq.empty))
```
Comment (crosslandwa) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/ClaimControllerSpec.scala:81]: OBSERVATION: I feel like we _should_ we verifying the correct "case reference number" and "conversation ID" are being passed to the `FileTransferService`

_(I'm happy for this PR to be merged with no action taken against this observation)_

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/ClaimControllerSpec.scala [hunk 4]
```diff
@@ -109,7 +107,7 @@ class ClaimControllerSpec extends ControllerSpec with GuiceOneAppPerSuite with T
           )
         )
 
-        when(mockFileTransferService.transfer(eisSuccessResponse.CaseID, uploads)).thenReturn(
+        when(mockFileTransferService.transferFiles(any(), any(), any())(any())).thenReturn(
```
Comment (crosslandwa) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/ClaimControllerSpec.scala:110]: OBSERVATION: As above, I feel like we _should_ we verifying the correct "case reference number" and "conversation ID" are being passed to the `FileTransferService`

_(I'm happy for this PR to be merged with no action taken against this observation)_

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**